docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,103 @@
+# Architecture
+
+An overview of the architecture for various components within the Toolkit.
+
+## Webviews
+
+The current implementation uses Vue 3 with Single File Components (SFCs) for modularity. Each webview
+is bundled into a single file and packaged into the toolkit at release time. Each component is able to
+act independently, however, they must respect the following principles:
+
+1. State can only be stored in a child component if it is not being used for two-way communication (via events)
+2. If there is two-way communication, store state in the parent
+3. Data should flow down, actions should flow up
```

Comment:
I'm not sure I agree with these two points (I'm assuming the "parent" is the "backend"?):
* 2: The frontend definitely needs a concept of state in order to help with persistence from VS Code. The frontend should be the source of truth for user inputs (and thus the state that the user has provided), and the backend should be able to handle inputs from the frontend functionally.
* 3: I would amend this to allow actions from elsewhere: actions don't need to be constrained to the webview if they allow data to flow down from the backend: the webview should be able to react in kind to any incoming payloads.

Reply:
"parent" in this context is a parent component, not the backend code. But yes the backend should not need to know about the frontend's state; the frontend will send a message when something needs to happen.

> the webview should be able to react in kind to any incoming payloads.

Agreed, and this is supported by the current implementation.